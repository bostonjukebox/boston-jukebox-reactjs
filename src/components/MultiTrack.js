import React from 'react'
import PropTypes from 'prop-types'

import styled from 'styled-components'

import Track from './Track'

import vocals from '../img/icon-vocals.svg'
import guitar from '../img/icon-guitar.svg'
import drums from '../img/icon-drums.svg'

const Container = styled.section`
    margin-top: 1em;
    display: flex;
    flex-direction: column;
    justify-content: center;
`

const MultiTrack = ({ playing }) =>
  <Container>
    <Track name="vocal" icon={vocals} playing={playing} />
    <Track name="guitar" icon={guitar} playing={playing} />
    <Track name="drums" icon={drums} playing={playing} />
  </Container>

MultiTrack.propTypes = {
  playing: PropTypes.bool,
}

export default MultiTrack
