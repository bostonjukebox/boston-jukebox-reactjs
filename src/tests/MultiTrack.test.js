import React from 'react'
import { shallow } from 'enzyme'
import MultiTrack from '../components/MultiTrack'

test('renders without crashing', () => {
  const component = shallow(<MultiTrack />)
  expect(component).toMatchSnapshot()
})

