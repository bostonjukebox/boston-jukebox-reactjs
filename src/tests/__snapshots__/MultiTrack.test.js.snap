// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.section>
    <Track
        icon="icon-vocals.svg"
        name="vocal"
        playing={undefined}
    />
    <Track
        icon="icon-guitar.svg"
        name="guitar"
        playing={undefined}
    />
    <Track
        icon="icon-drums.svg"
        name="drums"
        playing={undefined}
    />
</styled.section>,
  "nodes": Array [
    <styled.section>
      <Track
            icon="icon-vocals.svg"
            name="vocal"
            playing={undefined}
      />
      <Track
            icon="icon-guitar.svg"
            name="guitar"
            playing={undefined}
      />
      <Track
            icon="icon-drums.svg"
            name="drums"
            playing={undefined}
      />
</styled.section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <MultiTrack />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.section>
          <Track
                    icon="icon-vocals.svg"
                    name="vocal"
                    playing={undefined}
          />
          <Track
                    icon="icon-guitar.svg"
                    name="guitar"
                    playing={undefined}
          />
          <Track
                    icon="icon-drums.svg"
                    name="drums"
                    playing={undefined}
          />
</styled.section>,
        "_debugID": 2,
        "_renderedOutput": <styled.section>
          <Track
                    icon="icon-vocals.svg"
                    name="vocal"
                    playing={undefined}
          />
          <Track
                    icon="icon-guitar.svg"
                    name="guitar"
                    playing={undefined}
          />
          <Track
                    icon="icon-drums.svg"
                    name="drums"
                    playing={undefined}
          />
</styled.section>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <MultiTrack />,
}
`;
